compiler/passes/buildDefaultFunctions.cpp

```diff
@@ -641,6 +643,41 @@ static void build_record_inequality_function(AggregateType* ct) {
   normalize(fn);
 }
 
+
+static void build_enum_size_function(EnumType* et) {
+  if (function_exists("size", 1, et))
+    return;
+  // Build a function that returns the length of the enum specified
+  FnSymbol* fn = new FnSymbol("size");
+  fn->addFlag(FLAG_COMPILER_GENERATED);
+  fn->addFlag(FLAG_NO_PARENS);
+  fn->addFlag(FLAG_METHOD);
+  fn->insertFormalAtTail(new ArgSymbol(INTENT_BLANK, "_mt", dtMethodToken));
+  
+  ArgSymbol* _this = new ArgSymbol(INTENT_BLANK, "this", dtAny);//et);
+   _this->addFlag(FLAG_ARG_THIS);
+  _this->addFlag(FLAG_MARKED_GENERIC);
+  _this->addFlag(FLAG_TYPE_VARIABLE);
+  fn->insertFormalAtTail(_this);
```

Comment:
I believe you should add the _this ArgSymbol to the FnSymbol field of the same name.  That might help with the resolution problem you described in the PR message.